### PR TITLE
Sdk doc enhancements

### PR DESCRIPTION
**Pull request by szilveszterb**
### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,6 +43,8 @@ script:
             SDK_VERSION="2.0.0.dev${TRAVIS_BUILD_NUMBER}";
         fi;
         ./bin/build-sdk.sh $SDK_VERSION $DOC_VERSION;
+        docker build -t core:docs --target docs .;
+        docker tag core:docs core:dist;
```
Comment (ryansanford) [inline .travis.yml]: Prefer that Dockerfiles are autonomous and not require external setup and processing. Suggest branching off of this and trying:
  - Add a build stage in the Dockerfile above `dist` called `docs`
  - Have that build stage create the docs by emulating swagger/make.sh (FROM node:9-alpine,)
  - Have the `dist` build stage copy in the docs from the `docs` build stage (all dist images will have docs = good)
  - Above the line `./bin/build-sdk.sh $SDK_VERSION $DOC_VERSION;` copy the docs out of the `docs` image build stage for use by `build-sdk.sh`

I don't want to clutter this branch with this concern which may take some iteration.

**Comment by ryansanford** [comment]
Adding @ehlertjd as reviewer to make sure we don't violate any current or future expectations related to SDK CI.

**Comment by ehlertjd** [comment]
Hi @szilveszterb,

I just merged #1146 which will break a couple of items here, so I wanted you to be aware.

With that PR, the static content is expected to be 3 levels up from the generated docs. e.g.
```
├── branches
│   └── enhance-docs
│       ├── matlab
│       ├── python
│       └── swagger
└── static
    ├── images
    └── theme
        └── sphinx_rtd
            └── 0.3.0
                ├── css
                ├── fonts
                └── js
```

The static folder should be named the same, but the "branches" and "enhance-docs" folders can be named anything. I would suggest `flywheel/docs` or similar.

File: bin/copy_docs.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    dest_dir=$1
+
+    # Ensure that target exists
+    mkdir -p "${dest_dir}"
+
+    # Cleanup old docs, if necessary
+    rm -rf "${dest_dir}/python" "${dest_dir}/matlab"
+
+    # Python docs
+    cp -R sdk/src/python/sphinx/build "${dest_dir}/python"
```
Comment (ehlertjd) [inline bin/copy_docs.sh]: Please make sure we're doing the cleanup of unnecessary files (See the `#Cleanup sphinx folders` section from `push-docs.sh` [below](https://github.com/flywheel-io/core/pull/1165/files#diff-993a9e8ebaac46acc8c1ade31eea8b77L96). )

Also, please make sure that the static files get copied to `gh-pages` root, as in L87 and L94 of `push-docs.sh` below.

Suggest having separate functions to copy static files, and cleanup sphinx docs to be called at appropriate times.

**Comment by szilveszterb** [comment]
A lot of things changed since the first approach so I cleaned it up and opened a new pull request for it: https://github.com/flywheel-io/core/pull/1223

**Pull request closed without merging**